test/Constraints/diagnostics.swift

```diff
@@ -1111,3 +1111,6 @@ func elephant<T, U>(_: T) where T : Collection, T.Element == U, T.Element : Hash
 func platypus<T>(a: [T]) {
     _ = elephant(a) // expected-error {{generic parameter 'U' could not be inferred}}
 }
+
+// rdar://problem/34670592 - Compiler crash on heterogeneous collection literal
```

Comment:
This is not a diagnostics test since it does not diagnose, perhaps there is a better place we can put it?

Reply:
I was just thinking about that, moved it to `test/Constraints/construction.swift`